vllm/worker/model_runner_base.py

```diff
@@ -0,0 +1,205 @@
+import dataclasses
+from abc import ABC, abstractmethod
+from typing import (TYPE_CHECKING, Any, Dict, Generic, List, Optional, Tuple,
+                    Type, TypeVar, Union)
+
+import torch
+
+from vllm.sequence import SamplerOutput, SequenceGroupMetadata
+
+if TYPE_CHECKING:
+    from vllm.attention import AttentionMetadata
+    from vllm.attention.backends.abstract import AttentionBackend
+    from vllm.model_executor import SamplingMetadata
+
+T = TypeVar('T', bound="ModelInput")
+
+
+@dataclasses.dataclass(frozen=True)
+class ModelInput:
```

Comment:
because we previously use `ModelInput` for GPU input, let's rename this as `ModelInputBase`, so that people will not get confused in the future.